ert3/evaluator/_evaluator.py

```diff
@@ -1,25 +1,113 @@
 import asyncio
-import os
 import pathlib
 import shutil
 from collections import defaultdict
 from pathlib import Path
-from typing import Dict, Any, Tuple, List
+from typing import Coroutine, Dict, Any, Tuple, List
 
 
 import cloudpickle
-from pydantic import FilePath
 
 import ert3
 from ert_shared.ensemble_evaluator.config import EvaluatorServerConfig
 from ert_shared.ensemble_evaluator.evaluator import EnsembleEvaluator
 from ert_shared.ensemble_evaluator.prefect_ensemble import PrefectEnsemble
-
-from ert3.config import EnsembleConfig, StagesConfig, Step
-from ert3.data import EnsembleRecord, MultiEnsembleRecord, RecordTransmitter, Record
+from ert3.config import (
+    EnsembleConfig,
+    StagesConfig,
+    TransportableCommand,
+    Unix,
+    Function,
+    Step,
+)
+from ert3.data import (
+    EnsembleRecord,
+    MultiEnsembleRecord,
+    RealisationsToRecordToTransmitter,
+    RecordTransmitter,
+    Record,
+    SharedDiskRecordTransmitter,
+    record_data,
+)
 
 _EVTYPE_SNAPSHOT_STOPPED = "Stopped"
 _EVTYPE_SNAPSHOT_FAILED = "Failed"
+_MY_STORAGE_DIRECTORY = ".my_storage"
+
+CoroutineTransmitters = List[Coroutine[Any, Any, None]]
+
+
+# scafflting function. this will be removed later
+# with ert3.evaluator._evaluator._ee_config
+def _add_storage(path: Path) -> Path:
+    if path.name != _MY_STORAGE_DIRECTORY:
+        return path / _MY_STORAGE_DIRECTORY
+    return path
+
+
+def _create_input_transmitter(
+    name: str,
+    path: Path,
+    data: record_data,
+    futures: CoroutineTransmitters,
+) -> Dict[str, RecordTransmitter]:
+    transmitter = SharedDiskRecordTransmitter(name, _add_storage(path))
+    futures.append(transmitter.transmit_data(data))
+    return {name: transmitter}
+
+
+def _create_command_transmitter(
+    command: TransportableCommand,
+    path: Path,
+) -> RecordTransmitter:
+    transmitter = SharedDiskRecordTransmitter(command.name, _add_storage(path))
+    with open(command.path, "rb") as f:
+        asyncio.get_event_loop().run_until_complete(
+            transmitter.transmit_data([f.read()])
+        )
+    return transmitter
+
+
+def prepare_input(
+    step: Step,
+    evaluation_tmp_dir: Path,
+    ensemble_size: int,
+    inputs: MultiEnsembleRecord,
+) -> RealisationsToRecordToTransmitter:
+    (evaluation_tmp_dir / "prep_input_files").mkdir(parents=True, exist_ok=True)
+    futures: CoroutineTransmitters = []
+    transmitters = {
+        iens: _create_input_transmitter(
+            input_.name, evaluation_tmp_dir, record.data, futures
+        )
+        for input_ in step.inputs
+        for iens, record in enumerate(inputs.ensemble_records[input_.name].records)
+    }
```

Comment:
# transmitters = {
    #     iens: _create_input_transmitter(
    #         input_.name, evaluation_tmp_dir, record.data, futures
    #     )
    #     for input_ in step.inputs
    #     for iens, record in enumerate(inputs.ensemble_records[input_.name].records)
    # }

    transmitters = {}
    for input_ in step.inputs:
        for iens, record in enumerate(inputs.ensemble_records[input_.name].records):
            if iens in transmitters:
                transmitters[iens][input_.name] = _create_input_transmitter(
                    input_.name, evaluation_tmp_dir, record.data, futures
                )
            else:
                transmitters[iens] = {}
                transmitters[iens][input_.name] = _create_input_transmitter(
                    input_.name, evaluation_tmp_dir, record.data, futures
                )

Reply:
**Update**

Solution using default dicts:
```python 
    transmitters: Dict[int, Dict[str, RecordTransmitter]] = defaultdict(dict)

    for input_ in step.inputs:
        for iens, record in enumerate(inputs.ensemble_records[input_.name].records):
            transmitter = SharedDiskRecordTransmitter(input_.name, _add_storage(evaluation_tmp_dir))
            futures.append(transmitter.transmit_data(record.data))
            transmitters[iens][input_.name] = transmitter
```

The suggested solution assumes the removal of  the function`_create_input_transmitter`.

Also previous suggestion was not really working it was creating a dict like {0: {record_name: {record_name: Record}}} and it should have been {0: {record_name: Record}}